test/actions/katello/content_view_test.rb

```diff
@@ -416,5 +417,37 @@ class IncrementalUpdatesTest < TestBase
 
       assert_equal action.content_output(total_count), "with 3 Package(s), and 1 Errata"
     end
+
+    it 'runs' do
+      Dynflow::Testing::DummyPlannedAction.any_instance.stubs(:new_content_view_version).returns(::Katello::ContentViewVersion.first)
+
+      expected_cv_env_syncs = {}
+      expected_host_ids = []
+      expected_errata_ids = []
+
+      ::ForemanTasks.expects(:async_task).with(Actions::Katello::ContentView::IncrementalUpdate::SyncAndInstallContent,
+        {
+          cv_env_syncs: expected_cv_env_syncs,
+          host_ids: expected_host_ids,
+          errata_ids: expected_errata_ids
+        }
+      )
+
+      cvv_envs = [
+        {
+          content_view_version: content_view.version(library),
+          environments: [library]
+        }
+      ]
+
+      content = {
+        errata_ids: ['foo']
+      }
+
+      hosts = ::Host.all
+
+      planned = plan_action(action, cvv_envs, [], content, false, hosts, "", true)
+      run_action(planned)
```

Comment:
This test isn't fully built out yet but it's currently failing with this, and not sure what to make of it:

```
  1) Error:
Actions::Katello::ContentView::IncrementalUpdatesTest#test_0005_runs:
NoMethodError: undefined method `adapter' for nil:NilClass
    /home/vagrant/katello/test/actions/katello/content_view_test.rb:450:in `block in <class:IncrementalUpdatesTest>'
```

No errors in test.log, and this appears to mimic other tests that want to verify the run phase. Any ideas?

Reply:
so i tested this out locally and i think the problem is that the IncrementalUpdate action generates output that is passed to the IncrementalUpdates action into its plan_self method:
https://github.com/Katello/katello/blob/master/app/lib/actions/katello/content_view/incremental_updates.rb#L20-L23
Here's where it passes to plan_self
https://github.com/Katello/katello/blob/master/app/lib/actions/katello/content_view/incremental_updates.rb#L36

From what i can gather, this isn't actually supported to do within a test, at least not in the way that was executed here.  You could try calling ForemanTasks.sync_task(), as we do that in other places in tests to fully run an action and I don't remember seeing that issue there?  

You could also stub The 'IncrementalUpdate' action and provide some fake output, we do that in some places like here:   https://github.com/Katello/katello/blob/master/test/actions/katello/repository_test.rb#L243-L245